src/dali/plugin/pair_converter/csv/kraken.py

```diff
@@ -169,72 +171,135 @@ def _split_chunks_size_n(self, file_name: str, csv_file: str, chunk_size: int =
             elif position == _PAIR_START:
                 pair_start = int(chunk[0][self.__TIMESTAMP_INDEX])
 
-            chunk_filename: str = f'{pair}_{file_timestamp}_{duration_in_minutes}.{"csv.gz"}'
-            chunk_filepath: str = path.join(self.__CACHE_DIRECTORY, chunk_filename)
+            self._write_chunk_to_disk(pair, file_timestamp, duration_in_minutes, chunk)
 
-            with gopen(chunk_filepath, "wt", encoding="utf-8", newline="") as chunk_file:
-                csv_writer = writer(chunk_file)
-                for row in chunk:
-                    csv_writer.writerow(row)
+            # Emulate 1 week candle
+            if duration_in_minutes == _ONE_DAY:
+                week_chunk = []
+                for i in range(0, len(chunk), 7):
+                    seven_chunks = chunk[i : i + 7]
+
+                    # We want to make sure we have a full week for the averages and accurate pricing
+                    if len(seven_chunks) < 7:
+                        break
+
+                    # The timestamp of the first row becomes the timestamp for the weekly row
+                    col_sums: List[str] = [chunk[0][self.__TIMESTAMP_INDEX]]
+
+                    # We don't want/need to add up the timestamp column
+                    for col in range(self.__OPEN, self.__TRADES + 1):
+                        col_sum = str(sum((RP2Decimal(row[col]) for row in seven_chunks), ZERO))
+
+                        # Average all prices
+                        if col in range(self.__OPEN, (self.__CLOSE + 1)):
+                            col_sum = str(RP2Decimal(col_sum) / RP2Decimal("7"))
```

Comment:
You may want to use a different variable to store the average, e.g. column_average